### ISSUE TITLE
Indicate WebComponentsReady was dispatched in loader.js

### ISSUE THREAD

**Issue by hotforfeature** [webcomponents/webcomponentsjs#746]
When using `webcomponents-loader.js` to dynamically load polyfills, the loader will immediately dispatch the WebComponentsReady event on `requestAnimationFrame()` if no polyfills are needed.

https://github.com/webcomponents/webcomponentsjs/blob/master/webcomponents-loader.js#L46

In an Angular CLI generated project, the logic to bootstrap the application comes after the event has fired.

main.ts
```ts
window.addEventListener('WebComponentsReady', () => {
  platformBrowserDynamic().bootstrapModule(AppModule);
});
```

The above won't work on the latest Chrome, since no polyfills are needed and the event dispatches before the bootstrap logic. It _will_ work on browsers that need polyfills, or when using `webcomponents-lite.js` since the polyfills are loaded asynchronously.

`webcomponents-loader.js` could set a flag on the window indicating that the WebComponentsReady event was dispatched.

```ts
function bootstrap() {
  platformBrowserDynamic().bootstrapModule(AppModule);
}

if (window.webComponentsReady) {
  // No polyfills needed, bootstrap immediately
  bootstrap();
} else {
  // Wait for polyfills to load
  window.addEventListener('WebComponentsReady', bootstrap);
}
```

At the moment, a simple workaround would be to set a flag ourselves.

index.html
```html
<head>
  <script>
    window.webComponentsReady = false;
    window.addEventListener('WebComponentsReady', function() {
      window.webComponentsReady = true;
    });
  </script>
  <script src="assets/bower_components/webcomponentsjs/webcomponents-loader.js"></script>
</head>
<body>
  <app-root>Loading...</app-root>
  <!-- Angular CLI appends main.bundle.js here, after webcomponents-loader.js has run -->
</body>
```

This is not a critical feature, since there's a simple workaround. However, it did raise the question of other potential frameworks that might suffer the same effect, bootstrapping themselves after the event was fired and not knowing if they should wait.

**Comment by valdrinkoshi** [comment]
The event is fired after a RAF, so you should be able to listen for it in the script at the end of `<body>`, e.g. http://jsbin.com/toqiruc/1/edit?html,output

**Comment by hotforfeature** [comment]
True, though when it comes to the framework (Angular in this case), the actual bootstrap occurs asynchronously after the RAF.

The point I'm trying to make is there are lots of JS frameworks. From Angular to React, Knockout to jQuery and even Bootstrap. Some of them may run into a similar issue as Angular, where the frontend framework "bootstraps" and initializes the app asynchronously.

The workaround is easy. It would just be convenient to have a way for asynchronous app starts to determine if `WebComponentsReady` has fired, or if polyfills are still loading. Something like a flag on the `window.WebComponents` object.

**Comment by valdrinkoshi** [comment]
From the example you provided I understood that the CLI appends the `main.bundle.js` as the last child of `<body>`. Can't the `main.bootstrap.js` be wrapped in something like:
```js
addEventListener('WebComponentsReady', () => {
  // contents of your current main.bundle.js
});
```
Arguably, WebComponents polyfill could expose `WebComponents.whenReady(callback)`, but I guess the point is to not "pollute" `window` with globals.

@sorvell @azakus WDYT?

**Comment by hotforfeature** [comment]
That is exactly what I'm doing. Unfortunately, there are several other bundles (polyfills, vendor, and styles) that are loaded before the main bundle. By the time execution reaches `main.bundle.js`, the event has already fired.

**Comment by hotforfeature** [comment]
This is now fixed with 68ef957, we can use `window.WebComponents.ready`
